src/lib/components/grid.tsx

```diff
@@ -46,16 +50,20 @@ export class ReactGrid extends Component<IGridProps, IGridState> {
 
   reflectNewRowsOrCols(
     nextDescriptors: Array<Partial<IRowColDescriptor>>,
-    dim: IGridDimension
+    dim: IGridDimension,
+    isCols: boolean
```

Comment:
Maybe it should just take an object: `{rows, cols}` for the first param rather than this switch?

Reply:
can you give a code snippet of what you mean? i didn't like this switch either. i normally like the logic to be identical between the two dimensions so its DRY, but this is a special case for columns

Reply:
```jsx
this.reflectNewRowsOrCols(
{
  rows: nextProps.rows,
  cols: nextProps.rows
},
{
  rows: this.grid.rows,
  cols: this.grid.cols
});
```

Reply:
Ya I get the signature change but then how to abstract the function implementation so we’re not repeating those lines for each of the object keys?

Reply:
https://github.com/gridgrid/grid/blob/master/src/modules/abstract-row-col-model/index.js#L7

Looks like the rowCol factory takes a `row` or `col`, could that be used?

https://github.com/gridgrid/grid/blob/0615dbaf07ae228ad1edde74a324e9fe17a584b2/src/modules/abstract-row-col-model/index.js#L397-L400

```ts
if (typeof dim.col === 'function') // do builder stuff
```

Else, maybe extract a helper function that _does_ take `isCol` and call it for the `row` and `col` keys?

Reply:
ya may as well use the col function on the dimension that's a good idea. still special casey but doesn't require a gross boolean. thanks for the suggestion